arkane/mainTest.py

```diff
@@ -0,0 +1,110 @@
+# Software is furnished to do so, subject to the following conditions:        #
+#                                                                             #
+# The above copyright notice and this permission notice shall be included in  #
+# all copies or substantial portions of the Software.                         #
+#                                                                             #
+# THE SOFTWARE IS PROVIDED 'AS IS', WITHOUT WARRANTY OF ANY KIND, EXPRESS OR  #
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,    #
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE #
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER      #
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING     #
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER         #
+# DEALINGS IN THE SOFTWARE.                                                   #
+#                                                                             #
+###############################################################################
+
+"""
+This script contains unit tests of the :mod:`arkane.main` module.
+"""
+
+import unittest
+import os
+import shutil
+
+import rmgpy
+from arkane import Arkane
+
+################################################################################
+
+
+class TestArkaneExamples(unittest.TestCase):
+    """
+    Run all of Arkane's examples, and report which one failed
+    """
+    @classmethod
+    def setUp(self):
+        """A function that is run ONCE before all unit tests in this class."""
+        self.base_path = os.path.join(os.path.dirname(os.path.dirname(rmgpy.__file__)), 'examples', 'arkane')
+        self.failed = []
+        self.example_types = ['species', 'reactions', 'networks']
+
+    def test_arkane_examples(self):
+        for example_type in self.example_types:
+            example_type_path = os.path.join(self.base_path, example_type)
+            examples = []
+            for (dirpath, dirnames, filenames) in os.walk(example_type_path):
+                examples.extend(dirnames)
+                break
+            for example in examples:
+                path = os.path.join(example_type_path, example)
+                arkane = Arkane(inputFile=os.path.join(path, 'input.py'), outputDirectory=path)
+                arkane.plot = True
+                arkane.execute()
+                with open(os.path.join(path, 'arkane.log'), 'r') as f:
+                    log = f.readlines()
+                for line in log[::-1]:
+                    if 'execution terminated' in line:
+                        break
+                else:
+                    self.failed.append([example_type, example + 'FreeRotor'])
+        error_message = 'Arkane example(s) failed: '
+        for type_name_tuple in self.failed:
+            error_message += '{1} in {0}; '.format(type_name_tuple[0], type_name_tuple[1])
```

Comment:
I think you can directly unpack the tuple (though it's actually a list) here: `for example_type, example_name in self.failed`